### PR TITLE
Student mgmt updates

### PR DESCRIPTION
**Pull request by jrm86**
Added several functions to addStudentMgmt.sql

### CODE REVIEW

File: src/db/addStudentMgmt.sql [hunk 1]
```diff
@@ -303,10 +310,17 @@ RETURNS TABLE("FirstName" VARCHAR(50),
...
-   RAISE WARNING 'Function not implemented';
-END
-$$ LANGUAGE plpgsql
+
+SELECT COALESCE(s.fname, ''),
+       COALESCE(s.mname, ''), 
+       COALESCE(s.lname, ''), 
+       schoolissuedid, email, year
+FROM student s
+WHERE TRIM(COALESCE(s.fname)) ILIKE TRIM($1) AND
```
Comment (afig) [inline src/db/addStudentMgmt.sql]: These `COALESCE` statements on lines 319-321 don't seem to be doing much. I'm guessing that they should have `''` as their second parameter?

File: src/db/addStudentMgmt.sql [hunk 2]
```diff
@@ -402,10 +421,17 @@ CREATE OR REPLACE FUNCTION assignMidtermGrade(student INT,
 RETURNS VOID
 AS
 $$
-BEGIN
-   RAISE WARNING 'Function not implemented';
-END
-$$ LANGUAGE plpgsql
+WITH T AS (
```
Comment (afig) [inline src/db/addStudentMgmt.sql]: I think it would be nice if the CTE had a more descriptive name.

File: src/db/addStudentMgmt.sql [hunk 3]
```diff
@@ -349,10 +361,13 @@ CREATE OR REPLACE FUNCTION getStudentIDByIssuedID(schoolIssuedID VARCHAR(50))
...
-   RAISE WARNING 'Function not implemented';
-END
-$$ LANGUAGE plpgsql
+
+SELECT s.id
+FROM student s
+WHERE TRIM(s.schoolissuediD) ILIKE TRIM($1);
+
+$$ LANGUAGE sql
+
```
Comment (afig) [inline src/db/addStudentMgmt.sql]: Extra new line on 368? (Might also choose to fix it during the review tomorrow.)

**Review by afig — COMMENTED** [review]
Testing out the review feature. There should be 3 suggestions made that are visible in-line with the source.

File: src/db/addStudentMgmt.sql (hunk 1 above)
Comment (jrm86) [reply src/db/addStudentMgmt.sql]: You are correct.  Intent was to account for a student without a middle name, etc.  Will be addressed in a commit, shortly.

File: src/db/addStudentMgmt.sql (hunk 2 above)
Comment (jrm86) [reply src/db/addStudentMgmt.sql]: Good catch.  Fixing.

File: src/db/addStudentMgmt.sql (hunk 3 above)
Comment (jrm86) [reply src/db/addStudentMgmt.sql]: Extra lines removed here and a couple other places.  In either case, a good thing to check over, overall, during the review tomorrow. [Edited to fix a typo]

**Comment by afig** [comment]
Thanks for the fixes so far @jrm86. Upon testing, it seems like there's some issues with the function behavior, but part of this is an issue at the design level of the function. We need to know which section to assign the grade to, but it is not specified as a parameter. Currently, the best we can do is assign a grade to all sections that the student is enrolled in with the current instructor. We can discuss this later today in-person.

**Comment by jrm86** [comment]
I believe I already have a fix for when we meet later.

**Review by afig — APPROVED** [review]
Looks good to merge. Will fully test functionality later.

**Comment by DasilvaB** [comment]
Lines 464 and 428 should be sql instead of plpgsql. Actually functions need Begin and End instead of being sql.

**Comment by jrm86** [comment]
@DasilvaB, thanks for pointing this out.  I meant for it to be plpgsql, but I did forget to use `BEGIN` and `END` in the statement.  Fixed!

**Review by DasilvaB — APPROVED** [review]
Looks good.

**Pull request merged**